### PR TITLE
Grab lessee from node attributes and code improvements

### PR DESCRIPTION
**Pull request by rd4398**
I have tried to improve the code by making some changes and focusing on reducing number of API calls. However, I am not sure whether they are right (unit tests and PR checks seem to pass) and whether they will cause issues to other pieces of code...

Please let me know if any other changes are required in the code...

### CODE REVIEW

File: esi_leap/api/controllers/v1/node.py [hunk 1]
```diff
@@ -66,33 +66,41 @@ def get_all(self):
 
         project_list = keystone.get_project_list()
         now = datetime.now()
+
+        offers = offer_obj.Offer.get_all({"status": "available"},
+                                         context)
+
+        f_leases = lease_obj.Lease.get_all({"status": [statuses.CREATED]},
```
Comment (tzumainn) [inline esi_leap/api/controllers/v1/node.py:73]: I think I would just call this "leases"

File: esi_leap/api/controllers/v1/node.py [hunk 2]
```diff
@@ -66,33 +66,41 @@ def get_all(self):
...
+                                           context)
+
+        f_lease_uuids = ' '
```
Comment (tzumainn) [inline esi_leap/api/controllers/v1/node.py:76]: I don't think you need this

File: esi_leap/api/controllers/v1/node.py [hunk 3]
```diff
@@ -66,33 +66,41 @@ def get_all(self):
...
+            for o in f_leases:
+                if o.resource_uuid == node.uuid:
+                    f_lease_uuids = f_lease_uuids.join([o.uuid])
 
             n = Node(name=node.name, uuid=node.uuid,
                      owner=keystone.get_project_name(node.owner, project_list),
                      future_offers=future_offers,
                      future_leases=f_lease_uuids)
+
+            n.lessee = keystone.get_project_name(node.lessee, project_list)
```
Comment (tzumainn) [inline esi_leap/api/controllers/v1/node.py:98]: You can move this up above in the Node declaration, right below the owner line.

**Review by tzumainn — CHANGES_REQUESTED** [review]
Looks like a good improvement in the code! Some suggestions in-line.

File: esi_leap/api/controllers/v1/node.py (hunk 2 above)
Comment (rd4398) [reply esi_leap/api/controllers/v1/node.py:76]: `File "/home/rohand/esi-leap/.tox/py3/lib/python3.10/site-packages/mock/mock.py", line 1346, in patched
    return func(*newargs, **newkeywargs)

      File "/home/rohand/esi-leap/esi_leap/tests/api/controllers/v1/test_node.py", line 50, in test_get_all
    data = self.get_json("/nodes")

      File "/home/rohand/esi-leap/esi_leap/tests/api/base.py", line 77, in get_json
    response = self.app.get(full_path,

      File "/home/rohand/esi-leap/.tox/py3/lib/python3.10/site-packages/webtest/app.py", line 324, in get
    return self.do_request(req, status=status,

      File "/home/rohand/esi-leap/.tox/py3/lib/python3.10/site-packages/webtest/app.py", line 646, in do_request
    self._check_status(status, res)

      File "/home/rohand/esi-leap/.tox/py3/lib/python3.10/site-packages/webtest/app.py", line 675, in _check_status
    raise AppError(

    webtest.app.AppError: Bad response: 400 Bad Request (not 200 OK or 3xx redirect for http://localhost/v1/nodes)
b'{"faultcode": "Client", "faultstring": "Invalid input for field/attribute future_leases. Value: \'[]\'. Wrong type. Expected \'<class \'str\'>\', got \'<class \'list\'>\'", "debuginfo": null}'

`
From stack trace I think it is on line 47 in tests/api/controllers/v1/test_node.py

Comment (tzumainn) [reply esi_leap/api/controllers/v1/node.py:76]: Okay! I'll take a look when you submit the updated PR.

Comment (tzumainn) [reply esi_leap/api/controllers/v1/node.py:76]: Oh, you already submitted the updated PR. I'll take a look now!

File: esi_leap/api/controllers/v1/node.py [hunk 4]
```diff
@@ -66,33 +66,47 @@ def get_all(self):
...
                     future_offers.append(offer.uuid)
                 elif offer.end_time >= now:
                     current_offer = offer
             future_offers = ' '.join(future_offers)
 
-            f_leases = lease_obj.Lease.get_all({"resource_uuid": node.uuid,
-                                                "status": [statuses.CREATED]},
-                                               context)
-            f_lease_uuids = ' '.join([o.uuid for o in f_leases])
+            f_lease_uuids = f_lease_uuids.join(
```
[the remark this answers is not in the document]
Comment (rd4398) [reply esi_leap/api/controllers/v1/node.py:94]: That works as well, I will change it!

[the remark this answers is not in the document]
Comment (tzumainn) [reply esi_leap/api/controllers/v1/node.py]: This was a good suggestion!

**Review by tzumainn — APPROVED** [review]

**Comment by tzumainn** [comment]
Looks good now - thanks!

**Pull request merged**